app/models/admin/collection.rb

```diff
@@ -298,6 +316,11 @@ def create_s3_dropbox_directory!
       self.dropbox_directory_name = name
     end
 
+    def destroy_s3_dropbox_directory!
+      objects = Aws::S3::Bucket.new(name: Settings.encoding.masterfile_bucket).objects(prefix: "/dropbox/#{dropbox_directory_name}")
+      objects.batch_delete!
```

Comment:
This could take a long time if the collection is large. IMO it should be in a background job to prevent blocking user.